terraform/eval_output.go

```diff
@@ -76,16 +79,20 @@ func (n *EvalWriteOutput) Eval(ctx EvalContext) (interface{}, error) {
 		}
 	}
 
-	// If it is a list of values, get the first one
-	if list, ok := valueRaw.([]interface{}); ok {
-		valueRaw = list[0]
-	}
-	if _, ok := valueRaw.(string); !ok {
-		return nil, fmt.Errorf("output %s is not a string", n.Name)
+	switch valueTyped := valueRaw.(type) {
+	case string:
+		mod.Outputs[n.Name] = valueTyped
+	case []interface{}:
+		mod.Outputs[n.Name] = valueTyped[0]
```

Comment:
The old behavior there is pretty confusing (when would a list make it to this context, and why would getting the first value be the right thing to do here?) - does this case still make sense w/ native lists being supported?